lib/unique_within_region_validator.rb

```diff
@@ -23,7 +23,8 @@ def validate_each(record, attribute, value)
       else
         record_base_class.where(record_base_class.arel_attribute(attribute).lower.eq(value.downcase))
       end
-    matches = matches.where(options[:scope] => record.public_send(options[:scope])) if options.key?(:scope)
+    args = Array.wrap(options[:scope]).each_with_object({}) { |scope,h| h[scope] = record.public_send(scope) }
+    matches = matches.where(args)
     unless matches.in_region(record.region_id).where.not(:id => record.id).empty?
```

Comment:
sorry - didn't take into account next line
```suggestion
    attr_values_match = Array.wrap(options[:scope]).each_with_object({}) { |scope, h| h[scope] = record.public_send(scope) }
    unless matches.in_region(record.region_id).where.not(:id => record.id).where(attr_values_match).empty?
```

naming is hard - not sure about that one